coordinator/src/main/scala/filodb.coordinator/QueryActor.scala

```diff
@@ -117,22 +113,20 @@ final class QueryActor(memStore: MemStore,
     val numSchedThreads = Math.ceil(config.getDouble("filodb.query.threads-factor")
                                       * sys.runtime.availableProcessors).toInt
     val schedName = s"$QuerySchedName-$dsRef"
-
-    val thFactory = new ThreadFactory {
-      def newThread(r: Runnable) = {
-        val thread = new Thread(r)
-        thread.setName(s"$schedName-${thread.getId}")
+    val exceptionHandler = new UncaughtExceptionHandler {
+      override def uncaughtException(t: Thread, e: Throwable): Unit =
+        logger.error("Uncaught Exception in Query Scheduler", e)
+    }
+    val threadFactory = new ForkJoinPool.ForkJoinWorkerThreadFactory {
+      def newThread(pool: ForkJoinPool): ForkJoinWorkerThread = {
+        val thread = ForkJoinPool.defaultForkJoinWorkerThreadFactory.newThread(pool)
         thread.setDaemon(true)
-        thread.setUncaughtExceptionHandler(
-          new UncaughtExceptionHandler {
-            override def uncaughtException(t: Thread, e: Throwable): Unit =
-              logger.error("Uncaught Exception in Query Scheduler", e)
-          })
+        thread.setUncaughtExceptionHandler(exceptionHandler)
+        thread.setName(s"$schedName-${thread.getPoolIndex}")
         thread
       }
     }
-    // TODO retaining old fixed size pool for now - later change to fork join pool.
-    val executor = new ScheduledThreadPoolExecutor(numSchedThreads, thFactory)
+    val executor = new ForkJoinPool( numSchedThreads, threadFactory, exceptionHandler, true)
```

Comment:
Is async mode correct? Does monix detect the use of FJP and rely on it for joining tasks?

Reply:
Javadoc.
     * @param asyncMode if true, establishes local first-in-first-out
     * scheduling mode for forked tasks that are never joined. This
     * mode may be more appropriate than default locally stack-based
     * mode in applications in which worker threads only process
     * event-style asynchronous tasks.  For default value, use {@code
     * false}.

Monix does fall into above category. Moreoever, monix schedulers keeps async = true.